src/ipa_pc/mod.rs

```diff
@@ -378,106 +378,126 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
+            // LC(C) = SUM ( lambda^i * z_i(x) * C_i )
+            for (index, &comm_dash_segment) in labeled_commitment.commitment().comm.iter().enumerate() {
+                if index == lc_comm.len() {
+                    lc_comm.push(G::zero());
+                }
+                lc_comm[index] = (lc_comm[index].into_projective()
+                    + &(comm_dash_segment.into_projective()
+                        .mul(&z_i_value)
+                        .mul(&cur_challenge)
+                    )
+                ).into_affine();
+            }
 
-            computed_batch_v = computed_batch_v + &(cur_challenge * &((v_i - &y_i) / &(point - x_i)));
+            lc_value += y_i * cur_challenge * z_i_value;
 
+            // lambda^i
             cur_challenge = cur_challenge * &lambda;
         }
 
-        let mut batch_values = vec![];
-        for labeled_commitment in commitments.iter() {
-            batch_values.push(batch_proof.batch_values[
-                *labels
-                    .get(labeled_commitment.label())
-                    .ok_or(Error::MissingEvaluation {
-                        label: labeled_commitment.label().to_string(),
-                    })?
-                ]
-            );
-        }
+        // z(x)
+        let z_x_value = z_polynomial.evaluate(x_point);
+
+        // LC(C) = SUM ( lambda^i * z_i(x) * C_i ) - z(x) * C
+        lc_comm = lc_comm.iter().enumerate().map(|(index, &lc_comm_segment)| {
+            let h_comm_segment = if batch_proof.h_comm.len() > index {
+                batch_proof.h_comm[index]
+            } else {
+                G::zero()
+            };
+            (lc_comm_segment.into_projective() + &h_comm_segment.mul(-z_x_value)).into_affine()
+        }).collect::<Vec<_>>();
 
-        // Reconstructed v value added to the check
-        batch_values.push(computed_batch_v);
+        // Expected open value added to the check
+        let batch_values = vec![lc_value];
 
-        // The commitment to h(X) polynomial added to the check
-        let mut commitments = commitments;
+        // The reconstructed commitment to the LC polynomial put to the check
         let labeled_batch_commitment = LabeledCommitment::new(
-            format!("Batch"),
-            Commitment { comm: batch_commitment.clone(), shifted_comm: None },
+            format!("LC"),
+            Commitment { comm: lc_comm.clone(), shifted_comm: None },
             None
         );
-        commitments.push(&labeled_batch_commitment);
+        let batch_commitments = vec![&labeled_batch_commitment]; // commitments;
 
         let proof = &batch_proof.proof;
         end_timer!(evals_time);
 
         let check_time = start_timer!(|| "Succinct check batched polynomial");
 
-        fs_rng.absorb(&to_bytes![batch_proof.batch_values].unwrap());
-
-        let check_poly = Self::succinct_check(vk, commitments, point, batch_values, proof, fs_rng)?;
+        let check_poly = Self::succinct_check(
```

Comment:
We're calling the multi-poly single-point verifier even if we have just one poly and one value for which verify the opening. The function batches over commitments and values, but since we have just one commitment and one value we don't need batching and we waste time in doing useless operations.